### PR TITLE
fix: forward falsy logits processor in mlxlm format_output_type

### LINKED ISSUE #1980: mlxlm: MLXLMTypeAdapter.format_output_type uses falsy check instead of identity-None check
`MLXLMTypeAdapter.format_output_type` (line 97) uses `if not output_type` instead of `if output_type is None`:

```python
# mlxlm.py:97
def format_output_type(self, output_type=None):
    if not output_type:   # wrong: discards any processor where bool(processor) is False
        return None
    return [output_type]
```

If a logits processor subclass defines `__bool__` returning `False`, the processor is silently discarded and generation proceeds unconstrained — no error raised. Every other model file uses the correct pattern (`transformers.py`, `vllm.py`, `llamacpp.py`, etc.).

**Minimum example:**

```python
class FalsyProcessor:
    def __bool__(self): return False

def format_output_type_buggy(output_type=None):
    if not output_type:       # exact mlxlm.py code
        return None
    return [output_type]

p = FalsyProcessor()
print(p is None)                        # False
print(format_output_type_buggy(p))     # None — processor silently dropped
```

**Fix:** replace `if not output_type:` with `if output_type is None:`, consistent with every other model backend.

### PR DESCRIPTION
**Pull request by serhiizghama**
Closes #1980. `MLXLMTypeAdapter.format_output_type` used `if not output_type` to decide whether a logits processor was supplied. That's a truthiness check, so any processor whose `__bool__` (or `__len__`) evaluates to `False` gets silently dropped and generation runs unconstrained — no error, just wrong output. Only `None` should mean "no constraint".

Switched to the identity check `if output_type is not None`, matching `transformers.py`, `llamacpp.py` and the other backends.

Added two tests that don't need Apple Silicon (they build the adapter with a mock tokenizer): one for the `None` case and one for a falsy processor. The falsy-processor test fails on the old code and passes on the fix.

### CODE REVIEW

**Comment by serhiizghama** [comment]
Closing in favour of #1987, which was opened first and covers the same fix.
